### PR TITLE
Update AutoCompleteCommandBox and edited event handler

### PR DESCRIPTION
**Pull request by Joanna-YJA**
## Description
Fix bug where use have to press enter key twice to execute a command. It also fix bug to allow for command navigation, by updating CommandBox.java to AutoCompleteCommandBox.java. It ensures the command prompt suggestion is anchored below text field. Lastly, it is removes excessive logging.

## Checks
- [] No errors when running the tests
- [] Build and checkstyle passes
- [] Run the app and ran 2 commands without failing
- [] Written the baseline test cases for the PR

## Changelog
- []
- []
- []

### CODE REVIEW

File: src/main/resources/view/AutoCompleTextField.css [hunk 1]
```diff
@@ -0,0 +1,25 @@
+.context-menu {
+
+    -fx-background-color: #e3f9ff;
+    -fx-background-radius: 0 6 6 6;
+
+}
+
+.menu-item {
+ -fx-font-size: 15;
+}
```
Comment (hcwong) [inline src/main/resources/view/AutoCompleTextField.css]: Nitpicking: line separator here

File: src/main/java/seedu/address/ui/listpanel/EntityListPanel.java [hunk 2]
```diff
@@ -49,7 +47,7 @@ public EntityListPanel(ObservableList<? extends Entity> entityList) {
                 prefix = PrefixType.M;
                 listView.setCellFactory(listView -> new MentorListViewCell());
             }
-            logger.info("EntityListView has prefix type: " + this.prefix);
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/listpanel/EntityListPanel.java]: Nice to remove the logger. was a bit verbose previously 👍

File: src/main/java/seedu/address/ui/AutoCompleteCommandBox.java [hunk 3]
```diff
@@ -356,4 +340,39 @@ private TextFlow getOtherTemplate(String suggestion) {
...
+    /**
+     * Handles the Enter button pressed event.
+     */
+    @FXML
+    public void handleCommandEntered() {
+        try {
+            commandExecutor.execute(getText());
+            setText("");
+        } catch (CommandException | ParseException | AlfredModelHistoryException e) {
+            // setStyleToIndicateCommandFailure();
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: can delete if not using

File: src/main/java/seedu/address/ui/AutoCompleteCommandBox.java [hunk 4]
```diff
@@ -124,14 +112,13 @@ public void changed(ObservableValue<? extends Boolean> observable, Boolean oldVa
      */
     private void populatePopup(List<String> finalSuggestionResults) {
         List<CustomMenuItem> menuItems = new LinkedList<>();
-        int maxSuggestion = 5;
+        int maxSuggestion = 4;
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: can make this final i guess, small matter

File: src/main/java/seedu/address/ui/AutoCompleteCommandBox.java [hunk 5]
```diff
@@ -14,24 +14,34 @@
...
 
 /**
- * Represents an AutoCompleteTextField that provide command suggestions as user
- * types in command inputs.
+ * The UI component that is responsible for receiving user command inputs.
  */
-public class AutoCompleteTextField extends JFXTextField {
+public class AutoCompleteCommandBox extends JFXTextField {
+
+    public static final String ERROR_STYLE_CLASS = "error";
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: This variable dosent seem to be used anywhere, could be wrong here tho

File: src/main/java/seedu/address/ui/AutoCompleteCommandBox.java [hunk 6]
```diff
@@ -96,11 +85,11 @@ public void changed(ObservableValue<? extends String> observableValue, String ol
 
                 populatePopup(finalSuggestionResults);
                 if (!commandsPopup.isShowing()) {
-                    commandsPopup.show(AutoCompleteTextField.this, Side.BOTTOM, 0, 0);
+                    commandsPopup.show(AutoCompleteCommandBox.this, Side.BOTTOM, 0, 0);
                 }
 
                 // Request focus on first item
-                //commandsPopup.getSkin().getNode().requestFocus();
+                // commandsPopup.getSkin().getNode().requestFocus();
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: Can delete i guess if you are not using

File: src/main/java/seedu/address/ui/AutoCompleteCommandBox.java [hunk 7]
```diff
@@ -42,31 +52,11 @@
...
-                KeyEvent upEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "", "", KeyCode.UP, false, false, true, false);
-                this.fireEvent(upEvent);
-            }
+        this.commandExecutor = commandExecutor;
+        this.setPromptText("What can I do for you?");
 
-            if (downCombo.match(event)) {
-                KeyEvent downEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "", "", KeyCode.UP, false, false, true, false);
-                this.fireEvent(downEvent);
-            }
```
Comment (hcwong) [inline src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: Just curious, why does this not work?

Comment (Joanna-YJA) [reply src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: Becuase the event is thrown to every node including itself, then thrown repeatedly, so it results in stack overflow error.

Comment (hcwong) [reply src/main/java/seedu/address/ui/AutoCompleteCommandBox.java]: Ah I see thanks 👍

**Pull request merged**